tests/Feature/Nova/SignatureResourceTest.php

```diff
@@ -5,23 +5,119 @@
 namespace Tipoff\Waivers\Tests\Feature\Nova;
 
 use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Tipoff\Authorization\Models\User;
 use Tipoff\Waivers\Models\Signature;
 use Tipoff\Waivers\Tests\TestCase;
 
 class SignatureResourceTest extends TestCase
 {
     use DatabaseTransactions;
 
-    /** @test */
-    public function index()
+    private const NOVA_ROUTE = 'nova-api/signatures';
+
+    /**
+     * @dataProvider dataProviderForIndexByRole
+     * @test
+     */
+    public function index_by_role(?string $role, bool $hasAccess, bool $canIndex)
     {
+        $user = User::factory()->create();
+        if ($role) {
+            $user->assignRole($role);
+        }
+
+        $this->actingAs($user);
+
         Signature::factory()->count(4)->create();
 
-        $this->actingAs(self::createPermissionedUser('view signatures', true));
+        $response = $this->getJson(self::NOVA_ROUTE)
+            ->assertStatus($hasAccess ? 200 : 403);
+
+        if ($hasAccess) {
+            $this->assertCount($canIndex ? 4 : 0, $response->json('resources'));
+        }
+    }
+
+    public function dataProviderForIndexByRole()
+    {
+        return [
+            'Admin' => ['Admin', true, true],
+            'Owner' => ['Owner', true, true],
+            'Executive' => ['Executive', true, true],
+            //'Staff' => ['Staff', true, true],
```

Comment:
@pdbreen If I put the Staff here I received 403. Any cloud why?

Reply:
Might be location-based. The other tests may fail later depending on if the signature belongs to the Staff's location.

Reply:
@wolfrednicolas  Location would only filter results, not deny access.  403 is denial of access.  This should be debugged.

But, signature is also a location based resource.  So, all of these tests need to change to be based on the location based nova test template (from tipoff/checkout).  Any time you see a resource that has an `indexQuery(...)` method that adds location filtering (like this 
 - https://github.com/tipoff/waivers/blob/76e3eb1b97755f9090cde32693c2caace39b692e/src/Nova/Signature.php#L39), the location based template must be used.  Also, whether or not the query should be filtered should change from a role test to a permission test, like this - https://github.com/tipoff/checkout/blob/15ce19453d35483097484890f8e1b1ac27ed8711/src/Nova/Order.php#L39

Reply:
@wolfrednicolas - just pulled down and ran the test w/Staff role.  I didn't get a 403, I got the following error 
```
Failed asserting that actual size 0 matches expected size 4.
 /home/vagrant/src/waivers/tests/Feature/Nova/SignatureResourceTest.php:37
 ```

This is definitely caused by not using the location aware testing template when its needed.

Reply:
@pdbreen thank you for the explanation, sorry for the lapsus mental and put the 403 error.... in the `Signature` model I dont see any relation directly with `Location`, ...`Signature` has a `Room` and the `Room` has the `Location`

Reply:
The end result is the `indexQuery` is applying location based filtering of the results, which requires the location aware test template.

Reply:
return $query->whereHas('room', function ($roomlocation) use ($request) {
            return $roomlocation->whereIn('location_id', $request->user()->locations->pluck('id'));
        });
@pdbreen in these case I have 2 questions

1)this case the relation is not directly, ..`Signature` with `Locations`... I have to create the first the `Room` with the `Locations` and then the `Signature` with the room right?

2) if so, this package doesnt call the escape-room package, so I have to include the dependency in the composer and add the EscapeRoomServiceProvider to the testCase.. right?

Reply:
In this case, the lack of `location_id` on Signature is more likely a result of it not being necessary before because there wasn't any package boundary preventing access from Signature to Room.  Personally, I would either 
* change this package so Signature includes a location_id directly, breaking the dependency chain on room **OR**
* remove location filtering in the Nova resource for Signature

I would **not** create a direct dependency on escape-room for this.

Up to @drewroberts  to make a call on how to proceed.  If neither of the above changes are pursued, then at bare minimum, the filtering based on room location needs to be conditional on rooms actually being installed/existing.  So, this test should be added prior to the `whereHas` query:

```
if (!nova('room')) {
   return $query;
}
```